webapp/channels/src/components/common/scrollbars.tsx

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useMergeRefs} from '@floating-ui/react';
+import React, {useCallback, useRef} from 'react';
+import SimpleBar from 'simplebar-react';
+
+import 'simplebar-react/dist/simplebar.min.css';
+import './scrollbars.scss';
+
+export type ScrollbarsProps = {
+    children: React.ReactNode;
+    color?: string;
+    onScroll?: (e: React.UIEvent) => void;
+};
+
+const Scrollbars = React.forwardRef<HTMLElement, ScrollbarsProps>(({
+    children,
+    color,
+    onScroll,
+}, ref) => {
+    const removeListener = useRef<() => void>();
+
+    // We can't pass scroll handlers directly to SimpleBar, so we have to attach it to the DOM element directly
+    const setScrollRef = useCallback((el) => {
+        removeListener.current?.();
+        removeListener.current = undefined;
+
+        if (el && onScroll) {
+            el.addEventListener('scroll', onScroll);
+
+            removeListener.current = () => el.removeEventListener('scroll', onScroll);
+        }
+    }, [onScroll]);
+
+    const mergedRef = useMergeRefs<HTMLElement>([ref, setScrollRef]);
```

Comment:
Probably overkill, but we could use here the `useImperativeHandle`. That way, we can reduce and control the interface of this. That way, in many places where you need to do changes like `scrollToTop() -> scrollTo({top: 0})` we could just not make those changes, by exposing those things.

This way, if we ever change this library to something else, we just need to change the body of the imperative handler, and not everywhere in the code.

Reply:
I like that idea where if we were using those refs in more places, and if we needed to do more complicated things with them, but I generally prefer using the vanilla JS methods instead of using an imperative handle in cases like this